File: Application/VirgilSDKTests/VSS003_ModelsTests.m
//
//  VSS003_ModelsTests.m
//  VirgilSDK
//
//  Created by Oleksandr Deundiak on 10/13/16.
//  Copyright © 2016 VirgilSecurity. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <XCTest/XCTest.h>

#import "VSSTestsUtils.h"
#import "VSSCrypto.h"

@interface VSS003_ModelsTests : XCTestCase

@property (nonatomic) VSSTestsUtils * __nonnull utils;

@end

@implementation VSS003_ModelsTests

#pragma mark - Setup

- (void)setUp {
    [super setUp];
    
    self.utils = [[VSSTestsUtils alloc] initWithCrypto:[[VSSCrypto alloc] init]];
}

- (void)tearDown {
    [super tearDown];
}

#pragma mark - Tests

- (void)test001_CardImportExport {
    VSSCard *card = [self.utils instantiateCard];
    
    NSString *exportedData = [card exportData];
    
    VSSCard *importedCard = [[VSSCard alloc] initWithData:exportedData];
    
    XCTAssert([self.utils checkCard:card isEqualToCard:importedCard]);
}

- (void)test002_RevokeCardImportExport {
    VSSRevokeCard *revokeCard = [VSSRevokeCard revokeCardWithCardId:@"testId" reason:VSSCardRevocationReasonUnspecified];
    
    NSString *exportedData = [revokeCard exportData];
    
    VSSRevokeCard *importedRevokeCard = [[VSSRevokeCard alloc] initWithData:exportedData];
    
    XCTAssert([self.utils checkRevokeCard:revokeCard isEqualToRevokeCard:importedRevokeCard]);
}

@end

File: Application/VirgilSDKTestsSwift/VSS003_ModelsTests.swift
//
//  VSS003_ModelsTests.swift
//  VirgilSDK
//
//  Created by Oleksandr Deundiak on 10/17/16.
//  Copyright © 2016 VirgilSecurity. All rights reserved.
//

import Foundation
import XCTest

class VSS003_ModelsTests: XCTestCase {
    private var utils: VSSTestUtils!
    
    // MARK: Setup
    
    override func setUp() {
        super.setUp()
        
        self.utils = VSSTestUtils(crypto: VSSCrypto())
    }
    
    override func tearDown() {
        self.utils = nil
        
        super.tearDown()
    }
    
    func test001_CardImportExport() {
        let card = self.utils.instantiateCard()!
        
        let exportedData = card.exportData()
        
        let importedCard = VSSCard(data: exportedData)!
        
        XCTAssert(self.utils.check(card: card, isEqualToCard: importedCard))
    }

    func test002_RevokeCardImportExport() {
        let revokeCard = VSSRevokeCard(id: "testId", reason: .unspecified)
        
        let exportedData = revokeCard.exportData()
        
        let importedRevokeCard = VSSRevokeCard(data: exportedData)!
        
        XCTAssert(self.utils.check(card: revokeCard, isEqualToCard: importedRevokeCard))
    }
}
